src/node-zk.cpp

```diff
@@ -999,7 +999,9 @@ class ZooKeeper: public Nan::ObjectWrap {
 
             Nan::Utf8String _scheme (toString(toLocalVal(obj, LOCAL_STRING("scheme"))));
             Nan::Utf8String _auth (toString(toLocalVal(obj, LOCAL_STRING("auth"))));
-            uint32_t _perms = toUint(toLocalVal(obj, LOCAL_STRING("perm")));
+            // check if perm or perms exists, later can be changed to accept just 'perms' on next major release
```

Comment:
I would like to handle the "perm" and "perms" in the JavaScript code, instead of in the C++ AddOn code. Would you consider changing it into something similar to the Work in progress here? https://github.com/yfinkelstein/node-zookeeper/compare/master...DavidVujic:node-zookeeper:acl-perm-variations

Reply:
@DavidVujic Yes, I can do that.
My idea was to make it consistent on the addon level, not just JavaScript. Since it's inconsistent at the addon level I did a fix there. The next major release has perms consistent on every level, which can be easily achieved later by removing just the property check.

Reply:
@DavidVujic should we prioritize perms over perm if both are given for consistency?

Reply:
Good question! With the mapping of the acl code I think it will be dependent on the order of the keys. I think that is ok and good enough!